command/agent/dns.go

```diff
@@ -525,9 +538,17 @@ func syncExtra(index map[string]dns.RR, resp *dns.Msg) {
 // 1035. Enforce an arbitrary limit that can be further ratcheted down by
 // config, and then make sure the response doesn't exceed 512 bytes. Any extra
 // records will be trimmed along with answers.
-func trimUDPResponse(config *DNSConfig, resp *dns.Msg) (trimmed bool) {
+func trimUDPResponse(config *DNSConfig, req, resp *dns.Msg) (trimmed bool) {
```

Comment:
That API simplifies the method signature a bit, but it's quite easy to make an empty `dns.Msg`. It makes all the calls to `trimUDPResponse` a lot longer with more duplication to extract whether it's an edns message and pull out the max size. Each call prior to TrimUDPResponse has to parse out the req, which seemed like a lot of duplication.